Support Flatcar and provide ignition bootstrap data (#308)
ionos-cloud/cluster-api-provider-proxmox: docs/Usage.md

kubectl delete cluster proxmox-quickstart
```

#### Provision a cluster with Flatcar Container Linux Images

To provision a cluster with Flatcar, you need to build a suitable image with [image-builder](https://github.com/kubernetes-sigs/image-builder)
Once this PR [#1589](https://github.com/kubernetes-sigs/image-builder/pull/1589) is merged, you can follow the official docs on how to build a Flatcar image for Proxmox.
After you create a VM template using Flatcar, you can provision a cluster with the following options:

> ```shell

Author: Yeah you're right!

> ```shell

Author: it's done.

> ```shell

Author: it's done

kubectl delete cluster proxmox-quickstart
```

> #### Provision a cluster with flatcar

Reviewer: ```suggestion
#### Provision a cluster with Flatcar Container Linux Images
```

Reviewer: For people who have never heard of Flatcar to ensure they know it's about a different linux distribution

> To provision a cluster with flatcar,
> you need to build a suitable image with [image-builder](https://github.com/kubernetes-sigs/image-builder)
> once this PR is merged, https://github.com/kubernetes-sigs/image-builder/pull/1589
> please follow the docs in building flatcar image for Proxmox.
> 
> After having a VM template with flatcar, you can provision a cluster with the same environment variables:

Reviewer: ```suggestion
To provision a cluster with Flatcar, you need to build a suitable image with [image-builder](https://github.com/kubernetes-sigs/image-builder)
Once this PR [#1589](https://github.com/kubernetes-sigs/image-builder/pull/1589) is merged, you can follow the official docs on how to build a Flatcar image for Proxmox.

After you created a VM template using Flatcar, you can provision a cluster with the following options:
```

```shell
$ clusterctl generate cluster flatcar-quickstart \
    --infrastructure proxmox \
    --kubernetes-version v1.30.5 \
    --control-plane-machine-count 3 \
    --worker-machine-count 3 \
    --flavor flatcar > cluster.yaml

$ kubectl apply -f cluster.yaml
```

**Notes**: 

> - Make sure to define at least an ssh Key in the `VM_SSH_KEYS` environment variable, or the cluster will fail to provision.

Reviewer: ```suggestion
- Make sure to define at least one ssh key in the `VM_SSH_KEYS` environment variable, or the cluster will fail to provision.
```

- If you want more customization, you can extend the template to add multiple interfaces or dual-stack.

> - please always make sure that the ProxmoxMachines ignores the cloud-init status by defining `spec.checks.skipCloudInitStatus: true` in the ProxmoxMachine CR.

Reviewer: ```suggestion
- Please make sure that the ProxmoxMachines always ignore the cloud-init status by defining `spec.checks.skipCloudInitStatus: true` in the ProxmoxMachine CR.
```
